Support juju 3.4

rgildein opened the pull request:
Using master branch for zaza with TEST_JUJU3=1 [1] and dropping
bootstack-actions for func in CI to be able to use specific lxd
channel.

Changing to use of Juju 3.4 controller for testing instead of Juju 3.1, using TEST_JUJU3=1 [1] for zaza to install proper libjuju version. 

Simply sanitation of project and add workflow concurrency to avoid running multiple workflows on a single PR. Like this if new commit arrived the previous run will be canceled by new run.

~~blocked until [#655](https://github.com/openstack-charmers/zaza/pull/655) is not merged~~

---
[1]:
https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6

rgildein committed:
generic clean up

rgildein committed:
Drop bootstack-actions in CI func tests

Using master branch for zaza with TEST_JUJU3=1 [1] and dropping
bootstack-actions for func in CI to be able to use specific lxd
channel.

---
[1]:
https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6

rgildein committed:
using specific version of lxd

rgildein committed:
add CHARM_LOCATION

Pjack commented on .github/workflows/check.yaml:
```diff
@@ -22,19 +26,30 @@ jobs:
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    name: Functional tests
```
I would like to know the reason that why we cannot use worflows/func.yaml@v2 or v3? 
Do we have plan to retire worflows/func.yaml@v2 ?

Pjack commented on .github/workflows/check.yaml:
```diff
@@ -22,19 +26,30 @@ jobs:
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
+    name: Functional tests
```
I'm curious about the reason behind discontinuing the use of func.yaml. Is there a specific reason why we can't simply modify the existing one or introduce a new version? What's the major challenges to upgrading the controller? Initially, I had anticipated simply adjusting the 'juju-channel' value to 3.4.

My concern is that we may modify check.yaml again when we eventually upgrade to version 3.5. Minimizing these changes would be preferable.

Therefore, I prefer to know the whole plan and announce to whole team. thanks!

rgildein replied:
It was dou current version of those workflows does not support lxd-channel change and I hit the issue https://github.com/canonical/charmcraft/issues/1640. That issue was fixed.

My other reasons are:
- The reusable workflow for functional become too complex with too many parameters, so they are not readable what actually will be executed and how. If you open the pipeline, you see the lot of skipped steps, and you have no idea what actually was run.
- Any small change required us to change the workflow and then come back to the charm and if sometimes just created a new branch.
- We already have 4 branches, so it's not easy to change it on a single place.
- With potential Terraform use we can do it from one place, without that I would rather go and create 20 PRs with changing one parameter, than change it for 4 branches and then hope it works. 

This is an experiment and we can have discussion before merging any PR like this.

rgildein replied:
As we internally discuss and the issue https://github.com/canonical/charmcraft/issues/1640 was fixed, I'll revert this to using bootstack-actions again.

rgildein committed:
test of my fix for zaza

rgildein commented:
Converted to draft, to test zaza version.
Using my version [zaza](https://github.com/rgildein/zaza/tree/chore/no-ref/fix-action-object) to test fix for support 2.9 and 3.4 by zaza. Please do not merge.

rgildein committed:
fix requirements for install hook

Fix requirements for install hook not to failed, drop series from
metadata.yaml.

rgildein committed:
revert back usage of bootstack-actions and run only Juju 3.4

rgildein committed:
Switch to master branch of zaza

rgildein committed:
Add custom sshd config instead of changing value

The 60-cloudimg-settings.conf exist on new LXD with
"PasswordAuthentication no" so we need to define our own config file.

jneo8 reviewed (dismissed):
Please check why is the CI fail.

rgildein commented:
> Please check why is the CI fail.

It's due `PasswordAuthentication no`, which comes with new image configuration in `60-cloudimg-settings.conf`, and I tried to fix it with `'echo "PasswordAuthentication yes" > 01-test-settings.conf'` instead of `sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config`.
It was working on my local LXD fine, I'll test it again.

rgildein committed:
fix path to sshd config file

rgildein commented:
I did not use full path, so that was a problem.
